www/main/components/addons/sundar-gutka/components/SundarGutka.jsx

```diff
@@ -0,0 +1,92 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import anvaad from 'anvaad-js';
+
+import { Switch, Overlay } from '../../../../sttm-ui';
+import ExtraBani from './ExtraBani';
+import { convertToHyphenCase } from '../../../../utils';
+import { nitnemBaniIds, popularBaniIds } from '../../../../constants';
+
+import useLoadBani from '../hooks/use-load-bani';
+
+const SundarGutka = ({ isShowTranslitSwitch = false, onScreenClose }) => {
+  const { isLoadingBanis, banis } = useLoadBani();
+  const [isTranslit, setTranslitState] = useState(false);
+
+  const nitnemBanis = [];
+  const popularBanis = [];
+  const title = 'Sundar Gutka';
+  const hyphenedTitle = convertToHyphenCase(title.toLowerCase());
+  const overlayClassName = `ui-${hyphenedTitle}`;
+  const blockListId = `${hyphenedTitle}-banis`;
+  const blockListItemClassName = `${hyphenedTitle}-bani`;
+  const taggedBanis = banis.map(b => {
+    b.baniTag = '';
+
+    if (nitnemBaniIds.includes(b.id)) {
+      b.baniTag = 'nitnem';
+      nitnemBanis.push(b);
+    }
+    if (popularBaniIds.includes(b.id)) {
+      b.baniTag = 'popular';
+      popularBanis.push(b);
+    }
+
+    return b;
+  });
+
+  return (
+    <Overlay onScreenClose={onScreenClose}>
+      <div className={`${hyphenedTitle}-wrapper`}>
+        <div className={`bani-list overlay-ui ${overlayClassName}`}>
+          {isLoadingBanis ? (
+            <div className="sttm-loader" />
+          ) : (
+            <>
+              <header className="navigator-header">{title}</header>
+
+              {isShowTranslitSwitch && (
+                <Switch
+                  controlId="translit-switch"
+                  className="translit-switch"
+                  onToggle={setTranslitState}
+                  defaultValue={isTranslit}
+                />
+              )}
+
+              <section className="blocklist">
+                <ul id={blockListId} className="gurmukhi">
+                  {taggedBanis.map(b => (
```

Comment:
could this have been done in the first map?

Reply:
Yes, can be done but I wanted to separate the data from the component.